### PR TITLE
Gjoranv/container cluster implementations

### PR DESCRIPTION
**Pull request by gjoranv**

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/vespa/model/container/ContainerClusterImpl.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.yahoo.config.model.deploy.DeployState;
+import com.yahoo.config.model.producer.AbstractConfigProducer;
+
+/**
+ * The default container cluster implementation.
+ *
+ * @author gjoranv
+ */
+public final class ContainerClusterImpl
```
Comment (bjorncs) [inline config-model/src/main/java/com/yahoo/vespa/model/container/ContainerClusterImpl.java]: Consider renaming this to `DefaultContainerCluster` or `ApplicationContainerCluster` or similar. _Impl_ suffix give little information on what the class does.

File: config-model/src/main/java/com/yahoo/vespa/model/container/ContainerClusterImpl.java [hunk 2]
```diff
@@ -1,22 +1,145 @@
...
     }
 
     public ContainerClusterImpl(AbstractConfigProducer<?> parent, String subId, String name, ContainerClusterVerifier verifier, DeployState deployState) {
         super(parent, subId, name, verifier, deployState);
+        restApiGroup = new ConfigProducerGroup<>(this, "rest-api");
+        servletGroup = new ConfigProducerGroup<>(this, "servlet");
+    }
+
+
+    protected void myPrepare(DeployState deployState) {
```
Comment (bjorncs) [inline config-model/src/main/java/com/yahoo/vespa/model/container/ContainerClusterImpl.java]: Add `@Override` annotation. Consider renaming to `doPrepare` or similar.

File: config-model/src/main/java/com/yahoo/vespa/model/admin/LogserverContainerCluster.java [hunk 3]
```diff
@@ -12,9 +13,18 @@
...
+        addDefaultHandlersWithVip();
+        addLogHandler();
     }
 
     @Override
     protected void myPrepare(DeployState deployState) { }
 
+    private void addLogHandler() {
+        Handler<?> logHandler = Handler.fromClassName(ContainerCluster.LOG_HANDLER_CLASS);
+        logHandler.addServerBindings("http://*/logs", "https://*/logs");
```
Comment (bjorncs) [inline config-model/src/main/java/com/yahoo/vespa/model/admin/LogserverContainerCluster.java]: Use single binding `*://*/logs`.

**Review by bjorncs — APPROVED** [review]
See comments

**Review by hmusum — APPROVED** [review]
LGTM

**Review by bjorncs — APPROVED** [review]

**Review by baldersheim — APPROVED** [review]

**Pull request merged**